server/src/server.js

```diff
@@ -645,6 +645,14 @@ app.get("/images/:imageid", function (req, res) {
         anonAnalytics.pageview(analyticsUrl).send();
       }
       res.header("Content-Type", obj.contentType);
+      if (req.query.filename) {
+        let extension = "." + obj.contentType.split("/").pop();
+        let filename= req.query.filename;
+        if (! filename.endsWith(extension)) {
+          filename = filename + extension;
+        }
+        res.header("Content-Disposition", `attachment; filename="${filename}"`)
+      }
```

Comment:
This seems like a lot of trouble, and you have to pass around the signed value in the model, and maybe the links expire... can't we just give the content type to the page so it can construct a proper name?  Probably the type should be in the clip object itself, which would be fairly easy to add – right now it's always PNG, though I'd like to allow for JPEG eventually.